sql/moz-fx-data-shared-prod/monitoring_derived/bigquery_table_storage_timeline_daily_v1/query.py

```diff
@@ -0,0 +1,81 @@
+"""Determine big query table storage timeline per day."""
+
+from argparse import ArgumentParser
+
+from google.cloud import bigquery
+
+DEFAULT_PROJECTS = [
+    "mozdata",
+    "moz-fx-data-shared-prod",
+    "moz-fx-data-derived-datasets",
+    "moz-fx-data-marketing-prod",
+]
+
+parser = ArgumentParser(description=__doc__)
+parser.add_argument("--date", required=True)  # expect string with format yyyy-mm-dd
+parser.add_argument("--project", default="moz-fx-data-shared-prod")
+# projects queries were run from that access table
+parser.add_argument("--source_projects", nargs="+", default=DEFAULT_PROJECTS)
+parser.add_argument("--destination_dataset", default="monitoring_derived")
+parser.add_argument(
+    "--destination_table", default="bigquery_table_storage_timeline_daily_v1"
+)
+
+
+def create_query(date, source_project):
+    """Create query for a source project."""
+    return f"""
+
+
+        SELECT
+          "{source_project}" AS source_project,
+          DATE('{date}') AS change_date,
+          project_id,
+          table_schema AS dataset_id,
+          table_name AS table_id,
+          deleted,
+          DATE(creation_time) AS creation_date,
+          count(*) AS change_count,
+          avg(total_rows) AS avg_total_rows,
+          avg(total_partitions) AS avg_total_partitions,
+          avg(total_logical_bytes) AS avg_total_logical_bytes,
+          avg(active_logical_bytes) AS avg_active_logical_bytes,
+          avg(long_term_logical_bytes) AS avg_long_term_logical_bytes,
+          avg(total_physical_bytes) AS avg_total_physical_bytes,
+          avg(active_physical_bytes) AS avg_active_physical_bytes,
+          avg(long_term_physical_bytes) AS avg_long_term_physical_bytes,
+          avg(time_travel_physical_bytes) AS avg_time_travel_physical_bytes,
+          avg((active_logical_bytes/POW(1024, 3) * 0.02)
+          + ((long_term_logical_bytes/ POW(1024, 3)) * 0.01))
+          AS avg_logical_billing_cost_usd,
+          avg((active_physical_bytes/POW(1024, 3) * 0.04)
+          + ((long_term_physical_bytes/ POW(1024, 3)) * 0.02))
+          AS avg_physical_billing_cost_usd
+        FROM {source_project}.`region-us`.INFORMATION_SCHEMA.TABLE_STORAGE_TIMELINE
+        WHERE
+          DATE(timestamp) = '{date}'
+        GROUP BY 1, 2, 3, 4, 5, 6, 7
+        ORDER BY 1, 2, 3, 4, 5, 6, 7
```

Comment:
I recommend being explicit with the column names here, just in case someone adds another field in the future then it's easy to miss out on changing the numbers here.